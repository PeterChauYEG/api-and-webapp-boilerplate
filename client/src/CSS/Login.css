.FormContainer {
  position: fixed;
  right: 0;
  bottom: 0;
  left: 0;
  top: 0;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: flex-start;
}

.Login {
  height: 100vh;
  width: 100vw;
  position: absolute;
  left: 0;
  top: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
}