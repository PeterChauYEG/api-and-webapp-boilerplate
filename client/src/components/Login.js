import React, { Component } from 'react'

// import router
import { Redirect } from 'react-router-dom'

// import css
import '../CSS/Login.css'
// import 'font-awesome/css/font-awesome.css'

// import component
import HoverActionButton from './HoverActionButton'

class Login extends Component {
  constructor() {
    super()

    // bind functions
    this._handleNameChange = this._handleNameChange.bind(this)
    this._handlePasswordChange = this._handlePasswordChange.bind(this)

    // set initialState
    this.state = {
      credentials: {
        name: null,
        password: null,
      }
    }
  }

  // event handler for when Name changes
  // sets user.name state
  _handleNameChange(event) {

    // updates state
    this.setState({
      credentials: {
        ...this.state.credentials,
        name: event.target.value,
      }
    })

    return
  }

  // event handler for when Password changes
  // sets user.password state
  _handlePasswordChange(event) {

    // updates state
    this.setState({
      credentials: {
        ...this.state.credentials,
        password: event.target.value,
      }
    })

    return
  }

  _handleSubmit(authenticationRequest, credentials) {
    authenticationRequest(credentials)
    return
  }

  render() {
    // deconstruct props
    const {
      actions,
      location,
      user,
    } = this.props

    // deconstruct state
    const {
      credentials,
    } = this.state

    const { from } = location.state || '/'

    // const token = null
    console.log(location)
    
    return (
      <div>

        <div className="Login bg-light-blue">
          <div className="form-body">
            <h1>Login</h1>

            <div className="flex h-100 w-100 justify-between">
              <div className="flex flex-column justify-center w-100">

                {/* Name Input */}
                <div className="mv2">
                  <label
                    className="db"
                    htmlFor="name">
                    Name
                  </label>
                  <input
                    className="pa2 input-reset ba bg-transparent w-100 bw1 b--black"
                    type="text"
                    name="name"
                    id="name"
                    placeholder="anon"
                    onChange={this._handleNameChange} />
                </div>

                {/* Password Input */}
                <div className="mv2">
                  <label
                    className="db"
                    htmlFor="password">
                    Password
                  </label>
                  <input
                    className="pa2 input-reset ba bg-transparent w-100 bw1 b--black"
                    type="password"
                    name="password"
                    id="password"
                    placeholder="**********"
                    onChange={this._handlePasswordChange} />
                </div>
              </div>
            </div>
          </div>

          {/* Submit Button */}
          <HoverActionButton
            action={
              this._handleSubmit.bind(
                null,
                actions.authenticationRequest,
                credentials
              )
            }
            backgroundColor='bg-near-white'
            text='Submit'/>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  actions: React.PropTypes.object.isRequired,
}

export default Login;