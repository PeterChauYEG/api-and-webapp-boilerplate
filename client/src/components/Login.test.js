import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import Login from './Login'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const actions = {
  }
  const location = {
    state: null,
  }
  const user = {
    token: null,
  }
  
  // render component
  const component = renderer.create(
    <Login
      actions={actions} 
      location={location}
      user={user}
      />
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});
