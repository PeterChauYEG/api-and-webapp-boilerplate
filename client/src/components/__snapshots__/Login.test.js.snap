exports[`test renders without crashing 1`] = `
<div>
  <div
    className="Login bg-light-blue">
    <div
      className="form-body">
      <h1>
        Login
      </h1>
      <div
        className="flex h-100 w-100 justify-between">
        <div
          className="flex flex-column justify-center w-100">
          <div
            className="mv2">
            <label
              className="db"
              htmlFor="name">
              Name
            </label>
            <input
              className="pa2 input-reset ba bg-transparent w-100 bw1 b--black"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="anon"
              type="text" />
          </div>
          <div
            className="mv2">
            <label
              className="db"
              htmlFor="password">
              Password
            </label>
            <input
              className="pa2 input-reset ba bg-transparent w-100 bw1 b--black"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="**********"
              type="password" />
          </div>
        </div>
      </div>
    </div>
    <div
      className="HoverActionButton bg-near-white"
      onClick={[Function]}>
      <p>
        Submit
      </p>
    </div>
  </div>
</div>
`;
